src/hotspot/cpu/aarch64/vm_version_ext_aarch64.cpp

```diff
@@ -51,7 +51,28 @@ void VM_Version_Ext::initialize_cpu_information(void) {
   _no_of_threads = _no_of_cores;
   _no_of_sockets = _no_of_cores;
   snprintf(_cpu_name, CPU_TYPE_DESC_BUF_SIZE - 1, "AArch64");
-  snprintf(_cpu_desc, CPU_DETAILED_DESC_BUF_SIZE, "AArch64 %s", _features_string);
+
+  int fd = open("/proc/device-tree/compatible", O_RDONLY);
```

Comment:
This should not be done here in os-independent `src/hotspot/cpu/aarch64`. `src/hotspot/os_cpu/linux_aarch64` looks like a better place for this. Hotspot supports at least Windows/AArch64 in addition to Linux/AArch64.